src/components/store/AddCartButton/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { Dropdown } from '@/components/common';
+import { useEffect, useId, useState } from 'react';
+import styles from './style.module.scss';
+
+interface AddCartButtonProps {
+  currSize: string | undefined;
+  // Justification for disabling rules: This seems to be a false positive.
+  // https://stackoverflow.com/q/63767199/
+  // eslint-disable-next-line no-unused-vars
+  inStock: boolean;
+  inCart: boolean;
+  lifetimeRemaining: number;
+  montlyRemaining: number;
+  setInCart: (value: boolean) => void;
+}
+
+const AddCartButton = ({
+  currSize,
+  inStock,
+  inCart,
+  setInCart,
+  lifetimeRemaining,
+  montlyRemaining,
+}: AddCartButtonProps) => {
+  const [mounted, setMounted] = useState(false);
+  const [amount, setAmount] = useState<number>(1);
+
+  const myID = useId();
+
+  const maxCanBuy = Math.min(lifetimeRemaining, montlyRemaining);
+
+  useEffect(() => {
+    // console.log(`currSize ${currSize} inStock ${inStock} inCart ${inCart}`);
+    setMounted(true);
+  }, []);
+
+  if (!mounted) {
+    return null;
+  }
```

Comment:
https://www.npmjs.com/package/next-themes

I thought it would mess up the dark/light mode, but it seems like it won't since we don't use `useTheme()` specifically in this file. Removed it.